packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,645 @@
+      };
+
+      const consoleWarnSpy = jest
+        .spyOn(console, 'warn')
+        .mockImplementation(
+          () =>
+            'Some user actions could not be parsed. Please check the raw user actions data.'
+        );
+
+      mapAndSortActionsByMostRecent([invalidRawUserAction]);
+      expect(consoleWarnSpy).toHaveBeenCalledWith(
+        'Some user actions could not be parsed. Please check the raw user actions data.'
+      );
+
+      consoleWarnSpy.mockRestore();
+    });
+  });
+
+  describe('#splitActionsIntoTimelineSessions', () => {
+    describe('when it finds a case creation session', () => {
+      it('should properly split user actions into sessions and return the timeline including current session', async () => {
+        const mockRawActions = [...fakeActions];
+        const mappedAndSortedActions =
+          mapAndSortActionsByMostRecent(mockRawActions);
+        const ticketCreationDate = JSON.stringify(caseCreationDate.getTime());
+
+        const sessions = splitActionsIntoTimelineSessions(
+          mappedAndSortedActions,
+          ticketCreationDate
+        );
+
+        expect(sessions.session?.actions.length).toEqual(6);
+        expect(Number(sessions.session?.start)).toBeGreaterThan(
+          Number(
+            sessions.precedingSessions[sessions.precedingSessions.length - 1]
+              .end
+          )
+        );
+        expect(Number(sessions.session?.end)).toBeLessThan(
+          Number(sessions.followingSessions[0].start)
+        );
+
+        expect(sessions.precedingSessions.length).toEqual(1);
+        expect(sessions.precedingSessions[0].actions.length).toEqual(3);
+
+        expect(sessions.followingSessions.length).toEqual(2);
+        expect(sessions.followingSessions[0].actions.length).toEqual(2);
+      });
+    });
+
+    describe('when it does not find a case creation session', () => {
+      describe('when the ticket creation date comes before all the sessions', () => {
+        it('should return the timeline with the ticket creation date set as the current session timestamp and following sessions', async () => {
+          const mockRawActions = [...fakeActions].slice(0, 8);
+          const mappedAndSortedActions =
+            mapAndSortActionsByMostRecent(mockRawActions);
+          // Date far back before the first session
+          const ticketCreationDate = JSON.stringify(
+            createRelativeDate(firstSessionDate, -1000, 0).getTime()
+          );
+
+          const sessions = splitActionsIntoTimelineSessions(
+            mappedAndSortedActions,
+            ticketCreationDate
+          );
+
+          expect(sessions.session?.actions.length).toEqual(1);
+          expect(sessions.session?.actions[0].actionType).toEqual(
+            UserActionType.TICKET_CREATION
+          );
+          expect(sessions.session?.actions[0].timestamp).toEqual(
+            ticketCreationDate
+          );
+          expect(sessions).toEqual(sessions);
```

Comment:
We should remove that.

![image](https://github.com/user-attachments/assets/968e29b1-d2e9-4b47-8f10-99da9f93e1a2)

Reply:
I dont know what that was haha must have been tired :D